esphome/components/emporia_vue/emporia_vue.h

```diff
@@ -39,39 +39,48 @@ class CTSensor;
 
 class EmporiaVueComponent : public Component, public i2c::I2CDevice {
  public:
-  void setup() override;
-  void loop() override;
   void dump_config() override;
-  static void i2c_request_task(void *pv);
 
+  void set_sensor_poll_interval(uint32_t sensor_poll_interval) { this->sensor_poll_interval_ = sensor_poll_interval; }
+  uint32_t get_sensor_poll_interval() const { return this->sensor_poll_interval_; }
   void set_phases(std::vector<PhaseConfig *> phases) { this->phases_ = phases; }
   void set_ct_sensors(std::vector<CTSensor *> sensors) { this->ct_sensors_ = sensors; }
 
+  void setup() override;
+  void loop() override;
+
  private:
+  static void i2c_request_task(void *pv);
+
+  uint32_t sensor_poll_interval_;
   std::vector<PhaseConfig *> phases_;
   std::vector<CTSensor *> ct_sensors_;
   QueueHandle_t i2c_data_queue_;
 };
 
-enum PhaseInputWire : uint8_t { 
+enum PhaseInputWire : uint8_t {
   BLACK = 0,
   RED = 1,
-  BLUE = 2, 
+  BLUE = 2,
 };
 
 class PhaseConfig {
  public:
   void set_input_wire(PhaseInputWire input_wire) { this->input_wire_ = input_wire; }
-  void set_calibration(double calibration) {this->calibration_ = calibration; }
-  void set_voltage_sensor(sensor::Sensor voltage_sensor) {this->set_voltage_sensor = voltage_sensor; }
-  double get_calibration() { return this->calibration_; }
-  void update_voltage(EmporiaSensorData &data);
-  int32_t extract_power_for_phase(const PowerDataEntry &entry);
+  PhaseInputWire get_input_wire() const { return this->input_wire_; }
+  void set_calibration(float calibration) { this->calibration_ = calibration; }
+  float get_calibration() const { return this->calibration_; }
+  void set_voltage_sensor(sensor::Sensor *voltage_sensor) { this->voltage_sensor_ = voltage_sensor; }
+  sensor::Sensor *get_voltage_sensor() const { return this->voltage_sensor_; }
+
+  void update(const SensorReading &sensor_reading);
+
+  int32_t extract_power_for_phase(const ReadingPowerEntry &power_entry);
 
  private:
   PhaseInputWire input_wire_;
-  double calibration_;
-  sensor::Sensor voltage_sensor_;
+  float calibration_;
+  sensor::Sensor *voltage_sensor_{nullptr};
```

Comment:
I like this optional voltage sensor; I think we should do the same for power for now, and eventually current, frequency and period as well (maybe later though). Will include in a follow up PR

Reply:
Agreed. Power might be a little weird to put optional since that's the main sensor and the component doesn't really do anything without it.